### PR TITLE
use /usr/bin/env on long shebangs

### PR DESCRIPTION
**Pull request by kalefranz**
CC @ilanschnell @msarahan @mingwandroid 

Ok guys. This code works as expected.  But, what's going to happen on Windows? Should we make this a unix only thing?

### CODE REVIEW

File: conda/install.py [hunk 1]
```diff
@@ -357,6 +361,18 @@ def update_prefix(path, new_prefix, placeholder=prefix_placeholder,
     else:
         sys.exit("Invalid mode:" % mode)
 
+    if mode == 'text':
+        shebang_match = re.match(br'^(#!((?:\\ |[^ \n\r])+)(.*))', new_data)
+        if shebang_match:
+            whole_shebang, executable, options = shebang_match.groups()
```
Comment (kalefranz) [inline conda/install.py]: I couldn't help myself with this variable name...

**Comment by kalefranz** [comment]
P.S. I added some formatting fixes.  The real code is just the one section in install.py

**Comment by ilanschnell** [comment]
Windows does not understand shebangs anyway.  This is why we have the `.exe` proxy scripts on Windows.  So I would say: Yes, make this this a Unix only thing.

File: conda/install.py [hunk 2]
```diff
@@ -357,6 +361,18 @@ def update_prefix(path, new_prefix, placeholder=prefix_placeholder,
...
+            if len(whole_shebang) > 127:
+                executable_name = executable.split('/')[-1]
+                new_shebang = '#!/usr/bin/env {0}{1}'.format(executable_name, options)
+                print(whole_shebang, new_shebang)
```
Comment (msarahan) [inline conda/install.py]: debug line - just keep this in mind - I hate it when I forget them.

**Comment by msarahan** [comment]
I'm impressed.  I look forward to testing it.  Can you add a test for it in the test suite?

File: conda/install.py (hunk 2 above)
Comment (kalefranz) [inline conda/install.py]: thanks

File: conda/install.py [hunk 3]
```diff
@@ -338,45 +342,68 @@ def replace(match):
     assert len(res) == len(data)
     return res
 
-def update_prefix(path, new_prefix, placeholder=prefix_placeholder,
-                  mode='text'):
+
+def replace_long_shebang(mode, data):
+    if mode == 'text':
+        shebang_match = re.match(br'^(#!((?:\\ |[^ \n\r])+)(.*))', data)
```
Comment (kalefranz) [inline conda/install.py]: definitely not performant, but we can optimize later

**Pull request merged**